### DISCUSSION TITLE
Actions feedback

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#837] [Feature feedback]
Let's use this space to provide feedback for the Actions feature.

Post [an issue](https://github.com/avo-hq/avo/issues/new/choose) if something is broken.

**Comment by segiddins** [comment]
I just started writing the second action for RubyGems.org and I'm encountering some hiccups with actions that want to be able to reference other models. It seems like using `as: :belongs_to` gives me the selector I want in the action (it's based on one resource, and I want to take another model as an argument (or I guess field)), but then that field is filtered out in `BaseAction#handle_action` ... Basically, I'm wondering what the best way is to be able to add a selector for a model in an action _without_ it being directly related to the fields defined on the host resource

**Reply by adrianthedev** [reply]
Working on a solution for this.

**Reply by adrianthedev** [reply]
Hey @segiddins. Please try the new tags field in [2.26.2.pre.pr1579](https://rubygems.org/gems/avo/versions/2.26.2.pre.pr1579). It supports producing a single value (not an array of values) and doing Ajax requests to the server. So you can return whatever you like from the server and display the options in the dropdown.
You'll have access to the `record_id` and the `params` object.

Let me know if that works for you.

**Reply by adrianthedev** [reply]
See this documentation page (`mode` and `fetch_values_from`) for details on how to implement it.

https://docs.avohq.io/2.0/fields/tags.html#options

**Reply by segiddins** [reply]
https://github.com/rubygems/rubygems.org/pull/3637/files#diff-f3e2798eb6991f030d6adbc766214f03f31aef5e7afc996abc7e8997637f740e I ended up instead doing this

**Reply by adrianthedev** [reply]
You're really hacking this thing @segiddins. Great job! That's what I wanted folks to do.

Thanks for sharing!

**Comment by lorgio-jimenez** [comment]
Provide a reminder that if policies are being used, you need to add `act_on?` in the policy for it to be visible .

**Reply by adrianthedev** [reply]
Yes, I agree. Where do you think it would be best we add this reminder?
